### PR TITLE
feat(network): introduce Request.isNavigationRequest() method

### LINKED ISSUE #2627: How to set header in puppeter just main page request not all request
* Puppeteer version: 1.4
* Platform / OS version: ubuntu
* URLs (if applicable): any
* Node.js version: 8.x

I want to just apply header in just opened page in puppeter not all inside page headers

  page.setExtraHTTPHeaders({
     'X-Just-Must-Be-Request-In-Main-Request': '1',
  })

will apply all pages request headers all css/js/images and etc...

i want to set just main request headers not all of pages requests.

no one cares:
https://stackoverflow.com/questions/50568054/how-to-set-header-in-puppeter-just-main-page-request-not-all-request

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch introduces `Request.isNavigationRequest()` method.

Fixes #2627, #1132.

### CODE REVIEW

File: test/network.spec.js [hunk 1]
```diff
@@ -197,6 +197,17 @@ module.exports.addTests = function({testRunner, expect}) {
...
+    it('request.isNavigationRequest() should work', async({page, server}) => {
+      const requests = [];
+      page.on('request', request => requests.push(request));
+      server.setRedirect('/rrredirect', '/frames/one-frame.html');
+      await page.goto(server.PREFIX + '/rrredirect');
+      expect(requests[0].isNavigationRequest()).toBe(true);
+      expect(requests[1].isNavigationRequest()).toBe(true);
+      expect(requests[2].isNavigationRequest()).toBe(true);
+      expect(requests[3].isNavigationRequest()).toBe(false);
+      expect(requests[4].isNavigationRequest()).toBe(false);
```
Comment (JoelEinbinder) [inline test/network.spec.js]: How do we know the order of these? Looks flaky.

File: lib/NetworkManager.js [hunk 2]
```diff
@@ -249,7 +250,8 @@ class NetworkManager extends EventEmitter {
         redirectChain = request._redirectChain;
       }
     }
-    this._handleRequestStart(event.requestId, null, event.request.url, event.type, event.request, event.frameId, redirectChain);
+    const isNavigationRequest = event.requestId === event.loaderId && event.type === 'Document';
```
Comment (JoelEinbinder) [inline lib/NetworkManager.js:253]: The requestId can’t equal the loaderId and it not be a document request right? What happens when navigating to an image?

Comment (aslushnikov) [reply lib/NetworkManager.js:253]: The `request.resourceType()` is inferred by the request initiator rather then URL extension; so yes, it should always be `document` as far as I know. 

I'll add a test for navigation to the image to prove the point.

File: test/network.spec.js (hunk 1 above)
Comment (aslushnikov) [reply test/network.spec.js]: It's very deterministic:

- `requests[0]` is the initial request for the main document
- `requests[1]` is the redirected request for the main document
- `requests[2]` is the only request from the main document - the iframe main resource
- `requests[3]` and `requests[4]` are both subresources of the iframe.

**Review by JoelEinbinder — APPROVED** [review]

**Pull request merged**